### ISSUE TITLE
Constraint trait precedence for list/map/set's members

### ISSUE THREAD

**Issue by david-perez** [awslabs/smithy#1181]
The section on [constraint trait precedence](https://awslabs.github.io/smithy/1.0/spec/core/constraint-traits.html#trait-precedence) reads:

> Some constraints can be applied to shapes as well as **structure** members. If a constraint of the same type is applied to a **structure** member and the shape that the member targets, the trait applied to the member takes precedence.

(emphasis mine)

Note that the trait selectors for constraint traits allow them to be applied to _arbitrary_ members. For example, the [range trait](https://awslabs.github.io/smithy/1.0/spec/core/constraint-traits.html#range-trait)'s selector is `:test(number, member > number)`.

So this example model is accepted:

```smithy
$version: "1.0"

namespace com.amazonaws.simple

use aws.protocols#restJson1

@restJson1
service SimpleService {
    operations: [
        AnOperation,
    ],
}

@http(uri: "/operation", method: "GET")
operation AnOperation {
    input: AnOperationInputOutput,
    output: AnOperationInputOutput,
}

structure AnOperationInputOutput {
    list: MyList
}

list MyList {
    @range(min: 5)
    member: MyInt
}

@range(min: 1, max: 10)
integer MyInt
```

Does trait precedence apply in the case of a list member or a map's members, like in the above example?

**Comment by mtdowling** [comment]
Yeah this is just an error in the docs. It shouldn’t have the structure qualifier. It’s any member. Would you be able to send a PR to fix this?
